### PR TITLE
Create parent project POM

### LINKED ISSUE #220: Create parent project for PASS code repositories
Some concerns will be common to all PASS code repositories. This ticket is for creating a parent project for PASS repos to eventually  handle things like code style management and chscking dependencies

### PR DESCRIPTION
**Pull request by jabrah**
Resolves #220 

## Changes

* Add overarching project POM for PASS Java projects
  * This iteration: only enforce Duraspace checkstyle rules

## TODO

* ~~POM should be pushed somewhere that Maven can access it. GitHub's maven repository? Maven central?~~ (To be done in separate ticket #233)
* [x] Decide on proper `groupId`

### CODE REVIEW

**Comment by jabrah** [comment]
A few relevant questions here https://github.com/eclipse-pass/main/issues/230, seems like the values we have for `groupId` and `artifactId` should be fine for now. Other aspects like consolidating properties and dependencies are out of scope for this PR

**Comment by jabrah** [comment]
Should we leave the artifact publishing for a later PR?

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        <dependency>
+          <groupId>org.duraspace</groupId>
+          <artifactId>codestyle</artifactId>
+          <version>${duraspace-codestyle.version}</version>
+        </dependency>
+        <!-- Override dependencies to use latest version of checkstyle -->
+        <dependency>
+          <groupId>com.puppycrawl.tools</groupId>
+          <artifactId>checkstyle</artifactId>
+          <version>8.29</version>
```
Comment (bbranan) [inline pom.xml:83]: Could a more recent version of checkstyle be used?

**Comment by bbranan** [comment]
Follow-up ticket to handle publishing: https://github.com/eclipse-pass/main/issues/233

File: pom.xml (hunk 1 above)
Comment (jabrah) [reply pom.xml:83]: http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd
Suggests that we _should_ be able to run up to checkstyle v9.x, however, I was spot checking checkstyle plugin versions (manually changing the version in the IDEA plugin) and I'd get errors on any version above `8.41.1`. The error message suggested that the `checkstyle.xml` was invalid in versions `8.42+`

Error message in IDEA:
```
The Checkstyle rules file could not be parsed. cannot initialize module TreeWalker - cannot initialize module JavadocMethod - Property 'scope' does not exist, please check the documentation
```
Which looks like it refers to this part of the config: https://github.com/duraspace/codestyle/blob/master/src/main/resources/duraspace-checkstyle/checkstyle.xml#L98

Comment (jabrah) [reply pom.xml]: I've updated the checkstyle version to `8.41.1`, which also required an update to the maven-checkstyle-plugin to `3.1.2`

**Comment by jabrah** [comment]
We've updated checkstyle to the latest that is possible to run, given the Duraspace rules. So far, since we don't have a publishing plan yet, I've run `mvn install` locally for this repo to install the parent POM in my local m2 repository, allowing child projects to pull from there. Installing this parent POM works fine and running `mvn checkstyle:check` against one of the updated child projects also works.

**Review by bbranan — APPROVED** [review]

**Review by jrmartino — APPROVED** [review]
yep, this works

**Review by markpatton — APPROVED** [review]
Looks sensible. I'm able to install it.

**Pull request merged**